cpp/common/src/codingstandards/cpp/rules/identifierhidden/IdentifierHidden.qll

```diff
@@ -11,13 +11,36 @@ abstract class IdentifierHiddenSharedQuery extends Query { }
 
 Query getQuery() { result instanceof IdentifierHiddenSharedQuery }
 
-query predicate problems(UserVariable v2, string message, UserVariable v1, string varName) {
+/**
+ * There is a lambda that contains a declaration
+ * that hides something that is captured
+ * and the lambda exists in the function where this lamda is enclosed
+ */
+predicate hiddenInLambda(UserDeclaration v2, UserDeclaration v1) {
+  exists(Scope s, Closure le |
+    s.getADeclaration() = v2 and
+    s.getAnAncestor() = le and
+    le.getEnclosingFunction().getBasicBlock().(Scope) = v1.getParentScope() and
```

Comment:
Why are we looking at the basic block of the function? I believe this works because it coincides with the `BlockStmt` of the function.
Would it be better to look at the transitive parent scope of the closure or does that not have a parent scope that works?

Reply:
"Would it be better to look at the transitive parent scope..."
yeah that sounds more ideal but I wasnt able to get that to work 😞 

quick eval for `le.getParentScope()` has no result

tldr the scopes in lambdas werent as I naively expected, but if you see a better way to do this I could totally be missing something obvious!

Reply:
[your addition](https://github.com/knewbury01/codeql-coding-standards/pull/2) to this model deprecated this comment!